src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java

```diff
@@ -439,7 +442,25 @@ public void doTerm() {
             return;
         }
         final Throwable x = new FlowInterruptedException(Result.ABORTED);
-        Futures.addCallback(execution.getCurrentExecutions(/* cf. JENKINS-26148 */true), new FutureCallback<List<StepExecution>>() {
+        FlowExecution exec = getExecution();
+        if (exec == null) { // Already dead, just make sure statuses reflect that.
+            synchronized (getLogCopyGuard()) {
```

Comment:
Any possibility of some comments explaining the logic here?

Reply:
@abayer done!